From dbeab3700765d6e27662d13ebc21ba87b71b8307 Mon Sep 17 00:00:00 2001
From: mgcyung <mgcyung@gmail.com>
Date: Sat, 18 Jan 2020 14:36:47 +0800
Subject: [PATCH] What: gentoo-multilib-strict patch

Why:

How:
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9bafd06a..d64aaa08 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -909,7 +909,7 @@ add_library(Td::TdJson ALIAS TdJson)
 add_library(Td::TdJsonStatic ALIAS TdJsonStatic)
 
 install(TARGETS tdjson TdJson tdjson_static TdJsonStatic tdjson_private tdclient tdcore TdStatic EXPORT TdTargets
-  LIBRARY DESTINATION lib
+  LIBRARY DESTINATION lib64
   ARCHIVE DESTINATION lib
   RUNTIME DESTINATION bin
   INCLUDES DESTINATION include
-- 
2.24.1

